anonymize_public_ips.py

```diff
@@ -0,0 +1,111 @@
+    r"(?:" + IPV6SEG + r":){1,7}:",  # 1::                                 1:2:3:4:5:6:7::
+    r"(?:" + IPV6SEG + r":){1,6}:" + IPV6SEG,  # 1::8               1:2:3:4:5:6::8   1:2:3:4:5:6::8
+    r"(?:" + IPV6SEG + r":){1,5}(?::" + IPV6SEG + r"){1,2}",  # 1::7:8             1:2:3:4:5::7:8   1:2:3:4:5::8
+    r"(?:" + IPV6SEG + r":){1,4}(?::" + IPV6SEG + r"){1,3}",  # 1::6:7:8           1:2:3:4::6:7:8   1:2:3:4::8
+    r"(?:" + IPV6SEG + r":){1,3}(?::" + IPV6SEG + r"){1,4}",  # 1::5:6:7:8         1:2:3::5:6:7:8   1:2:3::8
+    r"(?:" + IPV6SEG + r":){1,2}(?::" + IPV6SEG + r"){1,5}",  # 1::4:5:6:7:8       1:2::4:5:6:7:8   1:2::8
+    IPV6SEG + r":(?:(?::" + IPV6SEG + r"){1,6})",  # 1::3:4:5:6:7:8     1::3:4:5:6:7:8   1::8
+    r":(?:(?::" + IPV6SEG + r"){1,7}|:)",  # ::2:3:4:5:6:7:8    ::2:3:4:5:6:7:8  ::8       ::
+    r"fe80:(?::"
+    + IPV6SEG
+    + r"){0,4}%[0-9a-zA-Z]{1,}",  # fe80::7:8%eth0     fe80::7:8%1  (link-local IPv6 addresses with zone index)
+    r"::(?:ffff(?::0{1,4}){0,1}:){0,1}[^\s:]"
+    + IPV4_ADDR_REGEX,  # ::255.255.255.255  ::ffff:255.255.255.255  ::ffff:0:255.255.255.255 (IPv4-mapped IPv6 addresses and IPv4-translated addresses)
+    r"(?:"
+    + IPV6SEG
+    + r":){1,4}:[^\s:]"
+    + IPV4_ADDR_REGEX,  # 2001:db8:3:4::192.0.2.33  64:ff9b::192.0.2.33 (IPv4-Embedded IPv6 Address)
+)
+IPV6_ADDR_REGEX = "|".join([f"(?:{g})" for g in IPV6GROUPS[::-1]])  # Reverse rows for greedy match
+
+
+def replace(filename, search_exp, replace_exp):
+    """Replace line when a substitution is needed."""
+    report = ""
+    try:
+        for line in fileinput.input(filename, inplace=True):
+            if any(pattern_to_skip in line for pattern_to_skip in PATTERNS_TO_SKIP):
+                newline = line
+            else:
+                newline = re.sub(search_exp, replace_exp, line)
+            sys.stdout.write(newline)
+            if line != newline:
+                if not report:
+                    report = f"IP Address found and replaced in {filename}\n"
+                report += f"  - {line}"
+    except UnicodeDecodeError as error:
+        return f"Warning: Not able to process {filename}: {error}"
+    return report
+
+
+def main():
+    """Main function."""
+    parser = argparse.ArgumentParser(description="Clean IP Addresses.")
+    parser.add_argument("--clean", action="store_true")
+    args = parser.parse_args()
+    clean = args.clean
+
+    ip_found = False
+    for (dirpath, _, files) in os.walk(TEST_DATA_PATH):
+        for file in files:
+            filename = os.path.join(dirpath, file)
+            if not clean:
+                report = ""
+                with open(filename, "r", encoding="utf-8") as test_file:
+                    try:
+                        content = test_file.readlines()
+                    except UnicodeDecodeError as error:
+                        print(f"Warning: Not able to process {filename}: {error}")
+                        continue
```

Comment:
sys.exit here? Or do we have files that we are *expecting* to have decoding errors in?

Reply:
yes, the CSV one is not supported, for now. We could solve it at some point if needed